### PR TITLE
Trying to improve cleanup - (Task #271)

### PR DESCRIPTION
**Pull request by ghost**
- Calling the clear editing domain registry from test setup and tear
down now.
- Changed registry clear to also shutdown NotificationThread.
- Changed code to wait for thread to be actually shutdown.

---
Task #271: Improve Cleanup of AProject test Case

### CODE REVIEW

**Review by andriikovalov-dlr — APPROVED** [review]

File: de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatTransactionalEditingDomain.java [hunk 1]
```diff
@@ -657,7 +657,16 @@ public void run() {
 		 */
 		public void finish() {
 			Activator.getDefault().getLog().log(new Status(Status.INFO, Activator.getPluginId(), "ResourceChangeEventThread: Thread triggered for stop.... "));
-			threadFinished = true;
+			triggerFinished = true;
+			
+			// Now wait until the thread actually terminated
+			while (this.isAlive()) {
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatTransactionalEditingDomain.java]: This looks like a re-implementation of the join() method of the Thread class.

File: de.dlr.sc.virsat.project.test/src/de/dlr/sc/virsat/project/test/AProjectTestCase.java [hunk 2]
```diff
@@ -96,8 +98,7 @@ public void tearDown() throws CoreException {
 		}
 
 		if (editingDomain != null) {
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.project.test/src/de/dlr/sc/virsat/project/test/AProjectTestCase.java]: Do we still need the null check? I think we can simplify this by always calling clear and setting the editingDomain to null regardless.

**Review by SaMuellerDLR — DISMISSED** [review]

File: de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatTransactionalEditingDomain.java (hunk 1 above)
Comment (PhilMFischer) [reply de.dlr.sc.virsat.project/src/de/dlr/sc/virsat/project/editingDomain/VirSatTransactionalEditingDomain.java]: Right. Done!

File: de.dlr.sc.virsat.project.test/src/de/dlr/sc/virsat/project/test/AProjectTestCase.java (hunk 2 above)
Comment (PhilMFischer) [reply de.dlr.sc.virsat.project.test/src/de/dlr/sc/virsat/project/test/AProjectTestCase.java]: Agree! Removed

**Review by franzTobiasDLR — APPROVED** [review]

**Pull request merged**